itemlist.lua

```diff
@@ -0,0 +1,231 @@
+--[[
+    show_item_list form for Minetest mods
+    (c) Pierre-Yves Rollo
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+--]]
+
+--[[ The item list dialog can only be displayed from within a node formspec.
+To open the item list dialog, add a do_file :
+
+local show_item_list = dofile(
+	minetest.get_modpath(minetest.get_current_modname())..'/itemlist.lua')
+
+...
+
+-- From your on_receive_field, call :
+show_item_list(player, title, item_list, pos_context)
+
+player: player name or object
+title: title to be displayed at the top of the dialog
+item_list: a list of item names
+pos_context: position of the context node (node that openned the formspec)
+]]--
+
+-- Formspec style
+
+local style={
+	linesize = 0.9, -- Line height
+	colsize = 4,    -- Column width
+	lines = 8,      -- Number of lines of items displayed
+	cols = 3,       -- Number of columns of items displayed
+}
+
+local modname = minetest.get_current_modname()
+
+-- Context management functions (surely many improvements to do)
+
+local contexts = {}
+
+local function get_player_name(player)
```

Comment:
Stable-0.4 builtin function:
https://github.com/minetest/minetest/blob/dd5f037/doc/lua_api.txt#L3020

Reply:
Hello SmallJocker and thanks for the comments :)

`is_player` is recent and I'm wondering if it is included in all servers versions (at least on ubuntu 18.04 LTS it is not available). What is the best to do in that case ?

Reply:
It's contained in 0.4.17.1 (stable-0.4 branch), the most recent Minetest release. Ubuntu also has PPAs available with detailed instructions on how to apply them in case the repository is outdated.
FYI: Other mods also make use of this function but I haven't seen any complaints yet.

Reply:
Ok, modification done in 1095b419ba51a08d272f65481f13edb76dd88a98